### ISSUE TITLE
Nested query doesn't work

### ISSUE THREAD

**Issue by cryogenian** [quasar-analytics/quasar#2821]
Problem accures in versions: `21.x.x`, `23.x.x` and `24.x.x` (didn't check `22.x.x`) 

First query is put to view mount 

```sql
select * from `/test-mount/testDb/flatViz`
```

The second one
```sql
SELECT row.seriesOne AS seriesOne, row.seriesTwo AS seriesTwo, MIN(row.measureOne) AS measureOne FROM `output_of_first_query` AS row GROUP BY row.seriesOne, row.seriesTwo ORDER BY row.seriesOne ASC, row.seriesTwo ASC
```
The third one is 
```sql
select * from `output_of_second_query` where seriesOne = "one-one" 
```

In `20.14.13` this works as expected in `>21.0.0` filter doesn't work. 

Interesting that if the second query is 
```sql
SELECT row.seriesOne AS seriesOne, MIN(row.measureOne) AS measureOne FROM `output_of_the_second_query` AS row GROUP BY row.seriesOne ORDER BY row.seriesOne ASC
```
this works.

**Comment by cryogenian** [comment]
Forgot to mention that output of the third query is like that query is just `select * from output_of_second_query`

**Comment by djspiewak** [comment]
Is this perhaps the subselect issue that's breaking charts, @wemrysi?

**Comment by alissapajer** [comment]
@djspiewak yes this is most likely that problem.

@cryogenian thanks for the report!

**Comment by alissapajer** [comment]
quasar v21.0.0 landed new mongo.

This is most likely a duplicate of #2775, #1929, and https://github.com/slamdata/slamdata/issues/2109.

**Comment by alissapajer** [comment]
Running
```sql
 select city as City, sport as Sport, min(year) as Min from olympics as row group by row.City, row.Sport order by row.City, row.Sport asc
```
as an example. This query returns multiple rows per `City`, `Sport` pair instead of the expected single row per pair.

Additionally, the previous query returns (with incorrect results) against mongo 3.4 in ~2.5 seconds. While the query
```
select city as City, sport as Sport, min(year) as Min from (select * from olympics) as row group by row.City, row.Sport order by row.City, row.Sport asc
```
returns against mongo 3.4 in 103 seconds.

**Comment by alissapajer** [comment]
Returns correct results in mongo 3.4:
```sql
select city, min(year) as Min from olympics group by city
```
Returns incorrect results in mongo 3.4 (full cross between the `min(year)` and the `City`:
```sql
select city as City, min(year) as Min from olympics group by City 
```
This bug is a duplicate of https://github.com/quasar-analytics/quasar/issues/2559. 

Furthermore I am happy to report that the query
```sql
select city, sport, min(year) as Min from olympics as row group by row.city, row.sport order by row.city, row.sport asc
``` 
returns correct results in mongo 3.4 as of PR #2902. I believe this would also return correct results on master but I did not verify that.

I am also happy to report that
```sql
select * from (select city, sport, min(year) as Min from (select * from olympics) as row group by row.city, row.sport order by row.city, row.sport asc) as foo where city = "Calgary"
```
returns correct results in ~2.0 seconds against mongo 3.4. This is equivalent to the original query reported as problematic. So this bug is officially a duplicate of #2559 and we can blame the SQL parser.

**Comment by alissapajer** [comment]
@cryogenian was this bug found running certain operations in SlamData? For example, while creating a chart or using other cards?

**Comment by alissapajer** [comment]
I just realized that my examples above were not exactly equivalent to the original examples we were grouping on the actual field names, while in my examples I was grouping on the field alias. The behavior I was seeing while grouping on the field alias is actually correct and what we would expect, because quasar currently does not support grouping on an alias.

I am re-running the queries to determine if I can, indeed, replicate the original issue.

**Comment by alissapajer** [comment]
@cryogenian I am not actually able to reproduce this bug as of quasar 25.0.7. Can I have some more information, maybe steps to take in SlamData?

I have determined the original query, through substitution, to be
```sql
select * from (SELECT row.seriesOne AS seriesOne, row.seriesTwo AS seriesTwo, MIN(row.measureOne) AS measureOne FROM (select * from `/test-mount/testDb/flatViz`) AS row GROUP BY row.seriesOne, row.seriesTwo ORDER BY row.seriesOne ASC, row.seriesTwo ASC) where seriesOne = "one-one" 
```
Since I don't have access to that data, I ran the nearly equivalent query on `olympics` (modulo the necessary `as foo`) on mongo 3.4
```sql
select * from (select row.city as seriesOne, row.sport as seriesTwo, min(row.year) as measureOne from (select * from olympics) as row group by row.city, row.sport order by row.city, row.sport asc) as foo where seriesOne = "Calgary"
```
This query returns correct results in <2 seconds.
```
 seriesOne | seriesTwo   | measureOne |
-----------|-------------|------------|
 Calgary   | Biathlon    | 1988       |
 Calgary   | Bobsleigh   | 1988       |
 Calgary   | Ice Hockey  | 1988       |
 Calgary   | Luge        | 1988       |
 Calgary   | Skating     | 1988       |
 Calgary   | Skiing      | 1988       |
```
As mentioned, there is an important difference between the two queries above, namely the addition of `as foo` in the `olympics` version. The original bug report did not include an alias for that `select`. Though since you're not actually inlining the queries when they are run, it wouldn't be necessary. But I wanted to point that out in case it was relevant.

It's also possible that this has been fixed since the issue was reported, or maybe I'm not replicating it correctly. I ran these tests against 45c2d4b87050fabb21bb5f078427b975bc4b3eb5, which is quasar version 25.0.7. 

@cryogenian if you need access to the latest quasar, let me know, since currently quasar publishing is not working.

**Comment by cryogenian** [comment]
This is fixed in `25.0.16`. Thank you!

**Comment by alissapajer** [comment]
I'm actually going to reopen this just so I can add an integration test for the query
```sql
select * from (select row.city as seriesOne, row.sport as seriesTwo, min(row.year) as measureOne from (select * from olympics) as row group by row.city, row.sport order by row.city, row.sport asc) as foo where seriesOne = "Calgary"
```
just to make sure we don't have any regressions.
